### PR TITLE
add event-based pagination [#171013108]

### PR DESCRIPTION
**Pull request by uraniumanchor**
# Contributing to the Donation Tracker

- [X] I've added tests or modified existing tests for the change.
- [X] I've humanly end-to-end tested the change by running an instance of the tracker.

### Issue from Pivotal Tracker

https://www.pivotaltracker.com/story/show/171013108

### Description of the Change

Certain pages ONLY scope for events, not just groups of 50. This adds two things to those pages:

- Show an event list if you try to view one of those pages without an event (it would redirect you to the latest event before)
- Show nav links above the table for any of these pages (previous, back to list, next)

### Verification Process

Created multiple events, made sure links showed up as expected (previous, next, list, etc).

### CODE REVIEW

File: tests/test_bid.py [hunk 1]
```diff
@@ -239,12 +239,12 @@ def test_donation_bid_admin(self):
 
 
 class TestBidViews(TestBidBase):
-    def test_bid_list(self):
-        resp = self.client.get(reverse('tracker:bidindex'))
-        self.assertRedirects(
-            resp, reverse('tracker:bidindex', args=(self.event.short,))
-        )
+    def test_bid_event_list(self):
+        resp = self.client.get(reverse('tracker:bidindex',))
```
Comment (yuna9) [inline tests/test_bid.py:243]: comma doesn't need to be there, I think

File: tracker/models/event.py [hunk 2]
```diff
@@ -207,6 +207,18 @@ class Event(models.Model):
     def __str__(self):
         return self.name
 
+    def next(self):
+        return (
+            Event.objects.filter(datetime__gte=self.datetime)
```
Comment (yuna9) [inline tracker/models/event.py:212]: i would prefer comparing to pk, but ig who cares

**Review by yuna9 — APPROVED** [review]
small stuff, seems fine

File: tracker/models/event.py (hunk 2 above)
Comment (uraniumanchor) [reply tracker/models/event.py:212]: Events are definitely meant to be ordered by start time (we've had this come up with GDQ, even) so this is very intentional.

Though now that I look at it again (I originally wrote this months ago) I'm wondering if there's a more elegant way of fetching this.

File: tests/test_bid.py (hunk 1 above)
Comment (uraniumanchor) [reply tests/test_bid.py:243]: Yeah, leftover from a copy paste. I'm surprised black didn't remove it, actually. Not important enough to fix, imo.

**Pull request merged**